### PR TITLE
feat(react-drawer): release to 9.0.0 stable

### PR DESCRIPTION
**Pull request by marcosmoura**
Release Drawer to stable. Drawer do not use the new `-preview` pattern.

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/Avatar-mount_base.html">633</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/Avatar-mount.html">643</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/Button-mount_base.html">308</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/Button-mount.html">297</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/Field-mount_base.html">1140</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/Field-mount.html">1103</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/FluentProvider-mount_base.html">703</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/FluentProvider-mount.html">718</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/FluentProviderWithTheme-mount_base.html">82</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/FluentProviderWithTheme-mount.html">84</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">64</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">68</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">72</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">79</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/MakeStyles-mount_base.html">850</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/MakeStyles-mount.html">858</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/Persona-mount_base.html">1678</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/Persona-mount.html">1688</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/SpinButton-mount_base.html">1313</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29711/perf-test-react-components/SpinButton-mount.html">1368</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `70.002 kB`<br />`20.161 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `207.417 kB`<br />`59.289 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |      `42.286 kB`<br />`14 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.651 kB`<br />`2.252 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/6ce4a1b8fb78294583f4906780f59b2d52ff3367'>6ce4a1b8fb78294583f4906780f59b2d52ff3367</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-drawer/package.json [hunk 1]
```diff
@@ -53,9 +53,7 @@
   },
   "beachball": {
     "disallowedChangeTypes": [
-      "major",
-      "minor",
-      "patch"
+      "major"
```
Comment (Hotell) [inline packages/react-components/react-drawer/package.json:56]:
```suggestion
      "major",
     "prerelease"
```

File: change/@fluentui-react-drawer-3cfc5c74-8b22-419c-a72a-43ecad9c876f.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "feat: release 9.0.0",
```
Comment (Hotell) [inline change/@fluentui-react-drawer-3cfc5c74-8b22-419c-a72a-43ecad9c876f.json:3]:
```suggestion
  "comment": "feat: release stable 9.0.0",
```

File: change/@fluentui-react-components-03e7f032-b0d9-4221-8522-b7bfb9ffb223.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "feat: release Drawer to stable",
```
Comment (Hotell) [inline change/@fluentui-react-components-03e7f032-b0d9-4221-8522-b7bfb9ffb223.json:3]:
```suggestion
  "comment": "feat: release Drawer as stable",
```

File: packages/react-components/react-components/src/index.ts [hunk 4]
```diff
@@ -1423,3 +1423,67 @@ export {
   useInfoLabel_unstable,
 } from '@fluentui/react-infolabel';
 export type { InfoLabelProps, InfoLabelSlots, InfoLabelState } from '@fluentui/react-infolabel';
+
+export {
```
Comment (Hotell) [inline packages/react-components/react-components/src/index.ts:1427]: we need to mark all `unstable` Drawer exports as deprecated ( `@deprecated` import drawer from `@fluentui/react-components`

**Review by layershifter — CHANGES_REQUESTED** [review]
We should not expose APIs from preview components in APIs for stable components:

https://github.com/microsoft/fluentui/blob/d35fc2628dc4b31acdb4e4ec9b6d5ff0c76860f3/packages/react-components/react-drawer/src/shared/DrawerBase.types.ts#L28
https://github.com/microsoft/fluentui/blob/d35fc2628dc4b31acdb4e4ec9b6d5ff0c76860f3/packages/react-components/react-drawer/src/shared/DrawerBase.types.ts#L32C39

Please update types to not reference `@fluentui/react-motion-preview`.

**Comment by layershifter** [comment]
> @layershifter Can you elaborate please?
> 
> * based on what rule is this statement. I'm not aware of any official agreement?
> * do we have any documentation for this?
> * drawer has `"@fluentui/react-motion-preview": "^0.4.0"` as direct dependency also it uses runtime apis
> 
> image
> * preview packages don't suffer semver breaking issues like old unstable releases

@Hotell the same discussion happened in #29391.

Currently, drawer components expose an API from preview package via `open` prop which is **public API**. You can see an example of its usage in https://react.fluentui.dev/?path=/docs/preview-components-drawer--default#motion-custom.

----

If a breaking change in `@fluentui/react-motion-preview` happens and modifies `MotionShorthand` or `MotionState`:
- it's fine to do breaks in `-preview`, we will a major bump of `@fluentui/react-motion-preview`
- we will get a break in public interface of `@fluentui/react-drawer` 🚨

However, now stable and preview packages linked by public interfaces. For example, I use `@fluentui/react-components` and `@fluentui/react-motion-preview` in a project:

```tsx
import { OverlayDrawer } from "@fluentui/react-components" // 9.0.0
import { useMotion } from "@fluentui/react-motion-preview" // @fluentui/react-motion-preview@0.4.0

function App() {
  const motion = useMotion();

  return <OverlayDrawer open={motion} />
}
```

A break in `@fluentui/react-motion-preview` happens, we bump `@fluentui/react-motion-preview` to 1.0.0 & `@fluentui/react-components` to 9.1.0 (as we depend on unstable API in `@fluentui/react-drawer`.


```tsx
import { OverlayDrawer } from "@fluentui/react-components" // 9.1.0
import { useMotion } from "@fluentui/react-motion-preview" // @fluentui/react-motion-preview@0.4.0

function App() {
  const motion = useMotion();

  // 💥💥💥💥💥💥💥
  // Builds breaks as `motion` type have been changed
  return <OverlayDrawer open={motion} />
}
```

---

It's fine to use `-preview` packages inside stable ones until we are starting to leak APIs from them. It have not been documented as it's a second time when it happens.

**Comment by Hotell** [comment]
thanks for providing needed context @layershifter , so basically what should be done here is to inline/copy that type within react-drawer component correct ?

**Comment by layershifter** [comment]
> so basically what should be done here is to inline/copy that type from react-motion-preview within react-drawer component correct ?

Yeah, that works.

@marcosmoura is there is a strict requirement to expose API for overrides? I mean this:

https://github.com/microsoft/fluentui/blob/d35fc2628dc4b31acdb4e4ec9b6d5ff0c76860f3/packages/react-components/react-drawer/src/shared/DrawerBase.types.ts#L28

Which is basically `boolean | Record<X, Y>`.

If there is no such requirement now, we will have less constraints in future if we will just expose `open: boolean` in props.

**Comment by Hotell** [comment]
> @marcosmoura is there is a strict requirement to expose API for overrides? I mean this:
> 
> https://github.com/microsoft/fluentui/blob/d35fc2628dc4b31acdb4e4ec9b6d5ff0c76860f3/packages/react-components/react-drawer/src/shared/DrawerBase.types.ts#L28
> 
> Which is basically `boolean | Record<X, Y>`.
> 
> If there is no such requirement now, we will have less constraints in future if we will just expose `open: boolean` in props.

yup if this would be ideal if possible 🙏

**Comment by marcosmoura** [comment]
> > so basically what should be done here is to inline/copy that type from react-motion-preview within react-drawer component correct ?
> 
> Yeah, that works.
> 
> @marcosmoura is there is a strict requirement to expose API for overrides? I mean this:
> 
> https://github.com/microsoft/fluentui/blob/d35fc2628dc4b31acdb4e4ec9b6d5ff0c76860f3/packages/react-components/react-drawer/src/shared/DrawerBase.types.ts#L28
> 
> Which is basically `boolean | Record<X, Y>`.
> 
> If there is no such requirement now, we will have less constraints in future if we will just expose `open: boolean` in props.

We can remove, but some partners need to do the overrides to integrate with their current code. We can remove the possibility for overrides right now, but those partners will need to hold a little while we release the motion package to stable as well.

**Comment by marcosmoura** [comment]
> We should not expose APIs from preview components in APIs for stable components:
> 
> https://github.com/microsoft/fluentui/blob/d35fc2628dc4b31acdb4e4ec9b6d5ff0c76860f3/packages/react-components/react-drawer/src/shared/DrawerBase.types.ts#L28
> 
> 
> https://github.com/microsoft/fluentui/blob/d35fc2628dc4b31acdb4e4ec9b6d5ff0c76860f3/packages/react-components/react-drawer/src/shared/DrawerBase.types.ts#L32
> 
> Please update types to not reference `@fluentui/react-motion-preview`.

@layershifter @Hotell 
I updated the types, and the Drawer code does not expose the Motion API anymore.

**Review by Hotell — APPROVED** [review]
🚢

**Comment by layershifter** [comment]
@marcosmoura after #29736 we still have:

https://github.com/microsoft/fluentui/blob/667d05dda55b7edbba8e41f483bf27f7dbefbc47/packages/react-components/react-drawer/src/shared/DrawerBase.types.ts#L31-L33

I mentioned it in the original message, too. Is there motivation to keep it as it? It's not so big issue as with `props`, but `*State` types are also a part of public API. Especially, if folks would use composition.

Can you please replace `MotionState` with `{ canRender: boolean; type: string }` (or something similar) to ensure that we will not have problems with breaking changes?

**Comment by Hotell** [comment]
> @marcosmoura after #29736 we still have:
> 
> https://github.com/microsoft/fluentui/blob/667d05dda55b7edbba8e41f483bf27f7dbefbc47/packages/react-components/react-drawer/src/shared/DrawerBase.types.ts#L31-L33
> 
> I mentioned it in the original message, too. Is there motivation to keep it as it? It's not so big issue as with `props`, but `*State` types are also a part of public API. Especially, if folks would use composition.
> 
> Can you please replace `MotionState` with `{ canRender: boolean; type: string }` (or something similar) to ensure that we will not have problems with breaking changes?

good catch, I checked only api.md report which while import from that package it doesn't show its leaking to anything public.

image

welp looks like we cannot rely on that thing :).

@marcosmoura let's refactor it to 


```diff
-import { MotionState } from '@fluentui/react-motion-preview';
type MotionState = { canRender: boolean; type: string }
export type DrawerBaseState = Required<Pick<DrawerBaseProps, 'position' | 'size'>> & {
-  motion: MotionState<HTMLDivElement>;
+  motion: MotionState;
};
```

though naked `type: string` is not the best type decision, because if we introduce strict union in future that would break users on TS level. 

image

While not breaking change on runtime, type checking would explode.  -> IMO its not breaking change rather than fix, but we dont provide any kinds of codemods etc which might provide not the best v9 bump for our users.

**Review by Hotell — APPROVED** [review]

**Review by layershifter — APPROVED** [review]

**Pull request merged**